### ISSUE TITLE
Cluster settings page sometimes requires multiple clicks to close

### ISSUE THREAD

**Issue by jim-docker** [lensapp/lens#2315]
Cluster settings page sometimes requires multiple clicks to close. It hasn't been reproduced in master but occurs for https://github.com/lensapp/lens/pull/2065 and https://github.com/lensapp/lens/pull/2047. May be related to using a pop-up `Menu` element for accessing the cluster settings page. The page may be getting added multiple times to the navigation history according to @Nachasic 

See the issue here:
https://github.com/lensapp/lens/pull/2065#pullrequestreview-592379673

**Comment by Nokel81** [comment]
This has to do with some odd behaviour around `navigation`. I don't know exactly what is causing it but from adding some inspection it is easy to observe that this behaviour only occurs when more than one cluster iframe is active.

Namely, adding the following code to `src/renderer/navigation/history.ts` can help observe what is going on.

```ts
navigation.listen((location, action) => {
  const clusterId = getHostedClusterId();
  const frame = clusterId ?? "root";

  logger.debug(`[NAVIGATION]: ${action}`, { location, frame });
});
```

**Comment by jim-docker** [comment]
fixed by #2330
